tests/platform/test_link_flap.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+import pytest
+
+from common.platform.device_utils import fanout_switch_port_lookup
+from common.utilities import wait_until
+
+class TestLinkFlap:
+    def __get_dut_if_status(self, dut, ifname=None):
+        if not ifname:
+            status = dut.show_interface(command='status')['ansible_facts']['int_status']
+        else:
+            status = dut.show_interface(command='status', interfaces=[ifname])['ansible_facts']['int_status']
+
+        return status
+
+
+    def __check_if_status(self, dut, dut_port, exp_state):
+        status = self.__get_dut_if_status(dut, dut_port)[dut_port]
+        return status['oper_state'] == exp_state
+
+
+    def __toggle_one_link(self, dut, dut_port, fanout, fanout_port):
+        logging.info("Testing link flap on {}".format(dut_port))
+
+        status = self.__get_dut_if_status(dut, dut_port)[dut_port]
+        logging.debug("Dut port status {}".format(status))
+        assert status['oper_state'] == 'up', "Skipping dut port {}: link operational down".format(status)
```

Comment:
this could reduced using the above helper function.

`assert self.__check_if_status(dut, dut_port, 'up'), '...'`

Reply:
True.

Down side is that moving the debug statement into the helper function would cause it come out every second instead of once after waiting is done.

Reply:
Found a good balance. Changed.